docs/apis/subsystems/task/adhoc.md

```diff
@@ -156,3 +156,32 @@ The `reschedule_or_queue_adhoc_task()` function will locate any existing task ma
 ```php
 \core\task\manager::reschedule_or_queue_adhoc_task($task);
 ```
+
+#### Set the retry limit for a task
+
+Since Moodle 4.4, the maximum number of retries for a task is **12** by default.
+
+In some situations you may wish to limit the number of times a task is retried. This can be accomplished using the `set_attempts_available()` function before queueing the task, for example:
+
+```php
+$task->set_attempts_available(2);
+\core\task\manager::queue_adhoc_task($task);
+```
+
+#### Get the remaining available attempts of a task
+Since Moodle 4.4, the maximum number of retries for a task is **12** by default and will be decreased by 1 each time the task is retried.
+
+You can get the remaining available attempts of a task by calling the `get_attempts_available()` function.
+
+```php
+$task->get_attempts_available();
+```
+
+#### No-retry tasks
+Since Moodle 4.4, tasks are retried by default. If you want to disable the retry mechanism for a task, you can override the `retry_until_success()` function and set it to false before queueing the task, for example:
```

Comment:
```suggestion

<Since version="4.4" issueNumber="MDL-79128" />

Since Moodle 4.4, tasks are retried by default. If you want to disable the retry mechanism for a task, you can override the `retry_until_success()` function and set it to false before queueing the task, for example:
```